app/src/main/java/com/github/studydistractor/sdp/eventChat/EventChatServiceFirebase.kt

```diff
@@ -0,0 +1,89 @@
+package com.github.studydistractor.sdp.eventChat
+
+import android.util.Log
+import com.github.studydistractor.sdp.data.FirebaseMessage
+import com.github.studydistractor.sdp.data.Message
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseException
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import java.util.Date
+
+class EventChatServiceFirebase : EventChatModel {
+    private val CHATPATH = "ChatEvent"
+
+    private val db = FirebaseDatabase.getInstance().getReference(CHATPATH)
+    private val auth = FirebaseAuth.getInstance()
+
+    private var currentChat : String = "test"
+
+    private var onChange : (List<Message>)-> Unit = {}
+
+    private val listener = object : ValueEventListener {
+        override fun onDataChange(snapshot: DataSnapshot) {
+            getMessages().continueWith{
+                t-> onChange(
+                    t.result
+                        .sortedWith(compareBy({it.timeStamp}))
+                )
+            }
+        }
+
+        override fun onCancelled(error: DatabaseError) {
+            Log.d("ERROR", error.message)
+        }
+    }
+
+    init {
+        db.child(currentChat).addValueEventListener(listener)
+    }
+
+    override fun messagesObserver(onChange: (List<Message>) -> Unit) {
+        this.onChange = onChange
+    }
+
+    override fun getMessages(): Task<List<Message>> {
+        return db.child(currentChat).get().continueWithTask{
+                t ->
+            if(t.isSuccessful){
+                val list = arrayListOf<Message>()
+                for(m in t.result.children){
+                    try {
+                        val message = m.getValue(FirebaseMessage::class.java)
+                        list.add(message!!.toMessage())
+                    } catch (_: DatabaseException){
```

Comment:
What about catching NullValueException?

Reply:
The issue is the getValue() it return DatabaseException if I'm correct.

Reply:
[Using the !! operator could result in a NullPointerException](https://kotlinlang.org/docs/null-safety.html), it would be best to catch it in order not to crash the app if the db doesn't contain the value.